### PR TITLE
Run test with config combination

### LINKED ISSUE #46: Run test configuration with combination.
In current implement below Configuration_1 will be treated as 2 test_configs (test_conf1 with pcrc and test_config2 with aggregation).
```
[Configuration_1]
pcrc=1
aggregation=1
```

This is not correct. Instead it shall be one test_config with pcrc+aggregation

### PR DESCRIPTION
**Pull request by mxu9**
Fix #46 

With below Configuration (default + pcrc)
```
[Configuration_1]
type            = link_ide
default         = 1
pcrc            = 1
```
The test will be like below:
```
[2024-04-24 00:11:58.547]  Print test results.
[2024-04-24 00:11:58.547]  TestSuite_1 - pass: 1, fail: 0, skip: 0
[2024-04-24 00:11:58.547]      TestGroup (LinkIDE Default+PCRC Test) - pass: 1, fail: 0, skip: 0
[2024-04-24 00:11:58.547]        Test.1: case - pass; ide_stream_secure - na
```

### CODE REVIEW

File: teeio-validator/teeio_validator/ide_test.c [hunk 1]
```diff
@@ -14,6 +14,17 @@
 #include "ide_test_group.h"
 #include "ide_test_case.h"
 
+#define LIST_APPEND(new, type)      \
```
Comment (jyao1) [inline teeio-validator/teeio_validator/ide_test.c:17]: Could you please use a function?

C macro is notorious for user mistake.

File: teeio-validator/teeio_validator/test_config/test_config_pcrc.c [hunk 2]
```diff
@@ -84,24 +83,18 @@ bool test_config_pcrc_disable_sel(void *test_context)
...
 {
-  return test_config_support_common(test_context) &&
-          test_config_check_pcrc_support_common(test_context);
+  return test_config_check_pcrc_support_common(test_context);
 }
 
 bool test_config_pcrc_check_sel(void *test_context)
 {
-  return test_config_check_common(test_context, "PCRC Config Assertion");
+  return true;
```
Comment (jyao1) [inline teeio-validator/teeio_validator/test_config/test_config_pcrc.c:91]: How PCRC is related to config combination?

Since it is a different issue, please split PCRC change to a different PR.

**Comment by mxu9** [comment]
Updated with PR #50
